src/linalg/impl_linalg.rs

```diff
@@ -293,8 +293,71 @@ impl<A, S, D> ArrayBase<S, D>
               A: LinalgScalar,
               E: Dimension,
     {
+        self.scaled_add_impl(alpha, rhs);
+    }
+
+    fn scaled_add_generic<S2, E>(&mut self, alpha: A, rhs: &ArrayBase<S2, E>)
+        where S: DataMut,
+              S2: Data<Elem=A>,
+              A: LinalgScalar,
+              E: Dimension,
+    {
         self.zip_mut_with(rhs, move |y, &x| *y = *y + (alpha * x));
     }
+
+    #[cfg(not(feature = "blas"))]
+    fn scaled_add_impl<S2, E>(&mut self, alpha: A, rhs: &ArrayBase<S2, E>)
+        where S: DataMut,
+              S2: Data<Elem=A>,
+              A: LinalgScalar,
+              E: Dimension,
+    {
+        self.scaled_add_generic(alpha, rhs);
+    }
+
+    #[cfg(feature = "blas")]
+    fn scaled_add_impl<S2, E>(&mut self, alpha: A, rhs: &ArrayBase<S2, E>)
+        where S: DataMut,
+              S2: Data<Elem=A>,
+              A: LinalgScalar,
+              E: Dimension,
+    {
+        debug_assert_eq!(self.len(), rhs.len());
+        assert!(self.len() == rhs.len());
```

Comment:
Right, the assertion there doesn't make any sense. With broadcasting, one stride would be `0` so that `equispaced_stride` returns `None` (as per the argument above), which would skip `axpy` and fall through to the default implementation. The assertion would simply prevent the broadcasting with the generic `scaled_add` to trigger.